src/servers/src/otlp/logs.rs

```diff
@@ -355,35 +358,343 @@ fn build_identity_row(
+                value_data: Some(ValueData::BoolValue(b)),
+            },
+            ColumnDataType::Boolean,
+            SemanticType::Field,
+            None,
+        ))),
+        JsonbValue::Array(_) | JsonbValue::Object(_) => OpenTelemetryNestValueIsComplicatedSnafu {
+            key: column_name.as_str(),
+        }
+        .fail(),
+        JsonbValue::Null => Ok(None),
+    };
+    column_info.map(|c| {
+        c.map(|(value, column_type, semantic_type, datatype_extension)| {
+            (
+                ColumnSchema {
+                    column_name,
+                    datatype: column_type as i32,
+                    semantic_type: semantic_type as i32,
+                    datatype_extension,
+                    options: None,
+                },
+                value,
+            )
+        })
+    })
+}
+
+#[derive(Debug)]
+enum OpenTelemetryLogRecordAttrType {
+    Resource,
+    Scope,
+    Log,
+}
+
+fn merge_schema<'a>(
+    extra_resource_schema: &'a SchemaInfo,
+    extra_scope_schema: &'a SchemaInfo,
+    extra_log_schema: &'a SchemaInfo,
+) -> BTreeMap<&'a String, (OpenTelemetryLogRecordAttrType, usize, &'a ColumnSchema)> {
+    let mut schemas = BTreeMap::new();
+    for (resource_schema_key, resource_schema_index) in extra_resource_schema.index.iter() {
+        if let Some(column_schema) = extra_resource_schema.schema.get(*resource_schema_index) {
+            schemas.insert(
+                resource_schema_key,
+                (
+                    OpenTelemetryLogRecordAttrType::Resource,
+                    *resource_schema_index,
+                    column_schema,
+                ),
+            );
+        }
+    }
+    for (scope_schema_key, scope_schema_index) in extra_scope_schema.index.iter() {
+        if let Some(column_schema) = extra_scope_schema.schema.get(*scope_schema_index) {
+            schemas.insert(
+                scope_schema_key,
+                (
+                    OpenTelemetryLogRecordAttrType::Scope,
+                    *scope_schema_index,
+                    column_schema,
+                ),
+            );
+        }
+    }
+    for (log_schema_key, log_schema_index) in extra_log_schema.index.iter() {
+        if let Some(column_schema) = extra_log_schema.schema.get(*log_schema_index) {
+            schemas.insert(
+                log_schema_key,
+                (
+                    OpenTelemetryLogRecordAttrType::Log,
+                    *log_schema_index,
+                    column_schema,
+                ),
+            );
+        }
+    }
+    schemas
+}
+
+fn parse_export_logs_service_request_to_rows(
+    request: ExportLogsServiceRequest,
+    select_info: Box<SelectInfo>,
+) -> Result<Rows> {
+    let mut schemas = build_otlp_logs_identity_schema();
+    let mut extra_resource_schema = SchemaInfo::default();
+    let mut extra_scope_schema = SchemaInfo::default();
+    let mut extra_log_schema = SchemaInfo::default();
```

Comment:
On second thought, can we use one `SchemaInfo` to collect resource, scope and log attributes in the same place?

Reply:
It might work, but the logic might be a little more complicated.

Reply:
We can open a separate PR to refactor this logic later.

Reply:
I can think of a situation that is not easy to deal with, that is, resource attr and log attr have the same name field but different types, and sharing a schema info will result in a type mismatch error when the first log attr is processed.

Reply:
All three schema lists will eventually be merged in `merge_schema` before the value is appended to the row, so even if they have the same name it's not going to matter. Also we don't differentiate selector keys with prefixes like `resource.key_name` or `scope.key_name`.

Reply:
The field with the same name, can not determine whether it is in other types of attr or its own type of attr, other types of attr in the field with the same name type can be overridden, but if it is in its own type of attr need to report an error, this may not be good to deal with.

Reply:
Yes, I suggest to use a single SchemaInfo to reduce allocate. The later ones should always override previous. Also I'm not sure if we can always use `String` for tag types.

Reply:
I think we can merge this pr first and follow up with a detailed discussion about merging schema info.